pokemon-showdown

```diff
@@ -22,6 +24,84 @@ try {
 	child_process.execSync('npm install --production', {stdio: 'inherit'});
 }
 
+// Check if the server is configured to use Go, and ensure the required
+// environment variables and dependencies are available if that is the case
+
+let config;
+try {
+	config = require('./config/config');
+} catch (e) {}
+
+if (config && config.golang) {
+	if (!process.env.GOPATH) {
+		console.log('The GOPATH environment variable is not set! It is required in order to run the server using Go.');
+		process.exit(0);
+	}
+	if (!process.env.GOROOT) {
+		console.log('The GOROOT environment variable is not set! It is required in order to run the server using Go.');
+		process.exit(0);
+	}
+
+	const dependencies = ['github.com/gorilla/mux', 'github.com/igm/sockjs-go/sockjs'];
+	let packages = child_process.execSync('go list all', {stdio: null, encoding: 'utf8'});
+	for (let dep of dependencies) {
+		if (!packages.includes(dep)) {
+			console.log(`Dependency ${dep} is not installed. Fetching...`);
+			child_process.execSync(`go get ${dep}`, {stdio: 'inherit'});
+		}
+	}
+
+	const {GOPATH} = process.env;
+	let stat;
+	let needsSrcDir = false;
+	try {
+		stat = fs.lstatSync(path.resolve(GOPATH, 'src/github.com/Zarel'));
```

Comment:
`GOPATH` is allowed to be a colon separated list (semicolon on Windows), in such an event, you may want to consider its first element. You can split the list by [`path.delimiter`](https://nodejs.org/api/all.html#path_path_delimiter)